lib/logstash/plugin_mixins/jdbc/statement_handler.rb

```diff
@@ -27,14 +27,30 @@ def post_init(plugin)
   class NormalStatementHandler < StatementHandler
     # Performs the query, respecting our pagination settings, yielding once per row of data
     # @param db [Sequel::Database]
-    # @param sql_last_value [Integet|DateTime|Time]
+    # @param sql_last_value [Integer|DateTime|Time]
+    # @param jdbc_paging_enabled [Boolean]
+    # @param jdbc_paging_manual_mode [Boolean]
+    # @param jdbc_page_size [Integer]
     # @yieldparam row [Hash{Symbol=>Object}]
-    def perform_query(db, sql_last_value, jdbc_paging_enabled, jdbc_page_size)
+    def perform_query(db, sql_last_value, jdbc_paging_enabled, jdbc_paging_manual_mode, jdbc_page_size)
```

Comment:
would be nice if there was a new class as it's a different algorithm e.g. `ExplicitStatementHandler` instead of hacking the algorithm into the existing one.

Reply:
nice idea, I come up with some commits to switch this